components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -390,12 +390,20 @@ impl Interpolate for BackgroundSize {
 impl Interpolate for RGBA {
     #[inline]
     fn interpolate(&self, other: &RGBA, progress: f64) -> Result<Self, ()> {
-        Ok(RGBA {
-            red: try!(self.red.interpolate(&other.red, progress)),
-            green: try!(self.green.interpolate(&other.green, progress)),
-            blue: try!(self.blue.interpolate(&other.blue, progress)),
-            alpha: try!(self.alpha.interpolate(&other.alpha, progress)),
-        })
+        let alpha = try!(self.alpha.interpolate(&other.alpha, progress));
+        alpha.min(1.);
```

Comment:
This doesn't do what you think (and is a no-op right now). `min` returns the value instead of modifying the underlying number.

So you need to write this as:

```rust
let alpha = try!(self.alpha.interpolate(&other.alpha, progress)).min(1.);
```
or

```rust
let alpha = cmp::min(1.0, try!(self.alpha.interpolate(&other.alpha, progress)));
```